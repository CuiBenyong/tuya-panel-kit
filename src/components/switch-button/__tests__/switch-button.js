import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import SwitchButton from '../index';

describe('SwitchButton Component', () => {
  it('basic render', () => {
    const component = renderer.create(<SwitchButton active={true} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('SwitchButton size&theme conflict', function() {
    const component = renderer.create(
      <SwitchButton
        value={true}
        active={true}
        size={{ height: 60 }}
        theme={{ width: 120, thumbSize: 55 }}
        onChange={data => {
          console.log(data);
        }}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('SwitchButton render width viewBox', function() {
    const onPress = jest.fn();

    const component = renderer.create(
      <SwitchButton
        value={true}
        active={true}
        onChange={data => {
          console.log(data);
        }}
      />
    );

    expect(component.root.instance.props.value).toBe(true);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
