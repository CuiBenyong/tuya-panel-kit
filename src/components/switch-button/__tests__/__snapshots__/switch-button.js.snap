// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchButton Component SwitchButton render width viewBox 1`] = `
<View
  style={
    Array [
      undefined,
      false,
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#44DB5E",
        "borderColor": "#44DB5E",
        "borderRadius": 16,
        "height": 32,
        "justifyContent": "center",
        "opacity": 1,
        "width": 52,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "#E5E5E5E5",
          "borderRadius": 14,
          "height": 28,
          "left": 22,
          "position": "absolute",
          "width": 28,
        }
      }
    />
  </View>
</View>
`;

exports[`SwitchButton Component basic render 1`] = `
<View
  style={
    Array [
      undefined,
      false,
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#44DB5E",
        "borderColor": "#44DB5E",
        "borderRadius": 16,
        "height": 32,
        "justifyContent": "center",
        "opacity": 1,
        "width": 52,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "#E5E5E5E5",
          "borderRadius": 14,
          "height": 28,
          "left": 22,
          "position": "absolute",
          "width": 28,
        }
      }
    />
  </View>
</View>
`;
